backend/src/main/java/com/github/tingshi2015/backend/reminder/ReminderService.java

```diff
@@ -46,8 +49,19 @@ public Reminder updateAReminder(ReminderDTO updateReminder, String id) {
         return reminderRepository.save(reminderToUpdate);
     }
 
+
 //    public Reminder getReminderById(String id) {
 //        return reminderRepository.findById(id)
 //                .orElseThrow(()-> new NoSuchElementException("Reminder with id: " + id + " not found. Can't getReminderById"));
 //    }
+
+    public List<Reminder> getUpcomingReminders() {
+        LocalDate today = LocalDate.now();
+        LocalTime now = LocalTime.now().withSecond(0).withNano(0); //ignore second & Nano-second
+
+        return reminderRepository.findAll().stream()
+                .filter(reminder -> reminder.date() != null && reminder.time() != null)
+                .filter(reminder -> reminder.date().equals(today) && reminder.time().withSecond(0).withNano(0).equals(now))
+                .collect(Collectors.toList());
+    }
```

Comment:
This is very inefficient. You request all reminders from the database and store them in the heap and then filter over them just to get some of them. This slows down your application and may crash when there are many reminders at some point by running out of memory. 
You should query your database directly with a method which only retrieves the necessary reminders, for example:

```
return reminderRepository.findRemindersByDateAndTime(today, now);
```

and in your repository your query:
```
@Query("{ 'date' : ?0, 'time' : ?1 }")
    List<Reminder> findRemindersByDateAndTime(LocalDate date, LocalTime time);
```

Reply:
ganz gut tip! danke sehr :)